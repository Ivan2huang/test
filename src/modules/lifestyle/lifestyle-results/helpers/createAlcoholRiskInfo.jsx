import React from 'react';

import { AlcoholIcon, NoAlcoholIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createAlcoholRiskInfo = (intl, theme, alcoholRisk) => {
  const alcoholRiskMapping = {
    None: `${theme.lifeStyleResultOK}`,
    RegularDrinker: `${theme.lifeStyleResultWarning}`,
    HeavyDrinker: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'alcohol',
    color: alcoholRiskMapping[alcoholRisk],
    CardIcon:
      alcoholRisk === 'None' ? (
        <NoAlcoholIcon />
      ) : (
        <AlcoholIcon color={alcoholRiskMapping[alcoholRisk]} />
      ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.alcoholRiskCard.title',
      'Alcohol',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.subTitle.${alcoholRisk}`,
      alcoholRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.alcoholRiskCard.description.${alcoholRisk}`,
      alcoholRisk,
    ),
  };
};

export default createAlcoholRiskInfo;
