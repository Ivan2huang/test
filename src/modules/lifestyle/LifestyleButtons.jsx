import React from 'react';
import { Box, withStyles } from '@material-ui/core';
import { injectIntl } from 'react-intl';
import * as PropTypes from 'prop-types';

import { EditIcon } from '../../icons';
import { formatMessage, navigateTo } from '../../helpers/helpers';
import paths from '../../helpers/paths';
import TrackingButton from '../../uiComponents/TrackingButton';
import { CATEGORIES } from '../../constants/analytics';

const Styles = theme => ({
  lifestyleBtn: {
    boxShadow: '0px 2px 6px rgba(0, 0, 0, 0.05)',
    background: theme.white,
    ...theme.typography['style-8'],
    padding: theme.spacingX(4),
    textAlign: 'left',
    borderRadius: theme.borderRadiusX(1),
    height: theme.spacingX(17),
    marginBottom: theme.spacingX(2),
    paddingRight: theme.spacingX(20),
  },
});

const LifestyleButtons = ({ intl, classes, questionnaireUpdateButton }) => {
  const navigateToQuestionnaire = () => navigateTo(paths.common.questionnaire);

  return (
    <>
      <TrackingButton
        className={classes.lifestyleBtn}
        data-testid="btn-questionnaire"
        onClick={navigateToQuestionnaire}
        trackingData={{
          category: CATEGORIES.MY_LIFESTYLE_OVERVIEW,
          action: questionnaireUpdateButton
            ? 'Update my lifestyle data'
            : 'Add my lifestyle data',
        }}
      >
        <EditIcon />
        <Box pl={4}>
          {formatMessage(
            intl,
            `lifestyle.buttons.${questionnaireUpdateButton ? 'update' : 'add'}`,
            'Add my lifestyle data',
          )}
        </Box>
      </TrackingButton>
    </>
  );
};

LifestyleButtons.defaultProps = {
  questionnaireUpdateButton: false,
};

LifestyleButtons.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  classes: PropTypes.exact({
    lifestyleBtn: PropTypes.string.isRequired,
  }).isRequired,
  questionnaireUpdateButton: PropTypes.bool,
};

export default withStyles(Styles)(injectIntl(LifestyleButtons));
