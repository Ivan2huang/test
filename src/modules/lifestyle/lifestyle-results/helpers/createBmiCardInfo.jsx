import React from 'react';

import Typography from '../../../../uiComponents/Typography';
import { formatMessage } from '../../../../helpers/helpers';

export const formatBmiScore = bmiScore => {
  const score = parseFloat(bmiScore);
  if (Number.isNaN(score)) {
    return '';
  }
  return (+bmiScore).toFixed(1);
};

const createBmiCardInfo = (intl, theme, bmiScore, bmiClass) => {
  const bmiColorMapping = {
    Healthy: `${theme.lifeStyleResultOK}`,
    Overweight: `${theme.lifeStyleResultWarning}`,
    Obese: `${theme.lifeStyleResultAlarm}`,
    Underweight: `${theme.lifeStyleResultWarning}`,
  };

  return {
    id: 'bmi',
    color: bmiColorMapping[bmiClass],
    CardIcon: (
      <Typography type="style-1" color={theme.grey3} fontWeight="bold">
        {formatBmiScore(bmiScore)}
      </Typography>
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.bmiCard.title',
      'BMI',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.subTitle.${bmiClass}`,
      bmiClass,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.bmiCard.description.${bmiClass}`,
      bmiClass,
    ),
  };
};

export default createBmiCardInfo;
