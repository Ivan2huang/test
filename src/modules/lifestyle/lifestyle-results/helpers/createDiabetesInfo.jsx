import React from 'react';

import { HeartIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createDiabetesInfo = (intl, theme, diabetesRisk) => {
  const alcoholRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    High: `${theme.lifeStyleResultWarning}`,
    VeryHigh: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'diabetes',
    color: alcoholRiskMapping[diabetesRisk],
    CardIcon: <HeartIcon color={alcoholRiskMapping[diabetesRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.diabetesCard.title',
      'Diabetes',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.subTitle.${diabetesRisk}`,
      diabetesRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.diabetesCard.description.${diabetesRisk}`,
      diabetesRisk,
    ),
  };
};

export default createDiabetesInfo;
