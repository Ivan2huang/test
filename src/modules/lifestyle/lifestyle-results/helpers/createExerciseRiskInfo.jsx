import React from 'react';

import { ExerciseIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createExerciseRiskInfo = (intl, theme, exerciseRisk) => {
  const exerciseRiskMapping = {
    High: `${theme.lifeStyleResultOK}`,
    LowToModerate: `${theme.lifeStyleResultWarning}`,
    VeryLow: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'exercise',
    color: exerciseRiskMapping[exerciseRisk],
    CardIcon: <ExerciseIcon color={exerciseRiskMapping[exerciseRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.exerciseRiskCard.title',
      'Exercise',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.subTitle.${exerciseRisk}`,
      exerciseRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.exerciseRiskCard.description.${exerciseRisk}`,
      exerciseRisk,
    ),
  };
};

export default createExerciseRiskInfo;
