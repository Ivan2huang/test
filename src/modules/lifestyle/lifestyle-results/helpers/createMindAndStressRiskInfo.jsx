import React from 'react';

import { StressIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createMindAndStressRiskInfo = (intl, theme, mindAndStressRisk) => {
  const mindAndStressRiskMapping = {
    Normal: `${theme.lifeStyleResultOK}`,
    BeCareful: `${theme.lifeStyleResultWarning}`,
    AtRisk: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'mindAndStress',
    color: mindAndStressRiskMapping[mindAndStressRisk],
    CardIcon: (
      <StressIcon color={mindAndStressRiskMapping[mindAndStressRisk]} />
    ),
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.mindAndStressRiskCard.title',
      'Mind & stress',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.subTitle.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.mindAndStressRiskCard.description.${mindAndStressRisk}`,
      mindAndStressRisk,
    ),
  };
};

export default createMindAndStressRiskInfo;
