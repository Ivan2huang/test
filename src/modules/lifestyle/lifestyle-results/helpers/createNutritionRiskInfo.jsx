import React from 'react';

import { NutritionIcon } from '../../../../icons';
import { formatMessage } from '../../../../helpers/helpers';

const createNutritionRiskInfo = (intl, theme, nutritionRisk) => {
  const nutritionRiskMapping = {
    Low: `${theme.lifeStyleResultOK}`,
    Moderate: `${theme.lifeStyleResultWarning}`,
    High: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'nutrition',
    color: nutritionRiskMapping[nutritionRisk],
    CardIcon: <NutritionIcon color={nutritionRiskMapping[nutritionRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.nutritionRiskCard.title',
      'Nutrition',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.nutritionRiskCard.subTitle.${nutritionRisk}`,
      nutritionRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.nutritionRiskCard.description.${nutritionRisk}`,
      nutritionRisk,
    ),
  };
};

export default createNutritionRiskInfo;
