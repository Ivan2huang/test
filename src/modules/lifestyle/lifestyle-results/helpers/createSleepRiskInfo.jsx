import React from 'react';

import { formatMessage } from '../../../../helpers/helpers';
import { SleepIcon } from '../../../../icons';

const createSleepRiskInfo = (intl, theme, sleepRisk) => {
  const sleepRiskMapping = {
    GoodSleeper: `${theme.lifeStyleResultOK}`,
    AverageSleeper: `${theme.lifeStyleResultWarning}`,
    PoorSleeper: `${theme.lifeStyleResultAlarm}`,
  };

  return {
    id: 'sleep',
    color: sleepRiskMapping[sleepRisk],
    CardIcon: <SleepIcon color={sleepRiskMapping[sleepRisk]} />,
    title: formatMessage(
      intl,
      'lifestyle.lifestyleResults.sleepRiskCard.title',
      'Sleep',
    ),
    subTitle: formatMessage(
      intl,
      `lifestyle.lifestyleResults.sleepRiskCard.subTitle.${sleepRisk}`,
      sleepRisk,
    ),
    description: formatMessage(
      intl,
      `lifestyle.lifestyleResults.sleepRiskCard.description.${sleepRisk}`,
      sleepRisk,
    ),
  };
};

export default createSleepRiskInfo;
