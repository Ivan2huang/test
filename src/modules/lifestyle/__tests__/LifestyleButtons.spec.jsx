/* eslint-disable react/prop-types */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import withTheme from '../../../themes/withThemeProvider';
import LifestyleButtons from '../LifestyleButtons';
import { navigateTo } from '../../../helpers/helpers';
import withIntl from '../../../i18n/withIntlProvider';

jest.mock('../../../helpers/helpers', () => ({
  navigateTo: jest.fn(),
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
}));

jest.mock('../../../helpers/paths', () => ({
  common: {
    findDoctor: '/clinic',
    questionnaire: '/questionnaire',
  },
}));

describe('LifestyleButton Component', () => {
  it('should match snapshot', () => {
    const Component = withIntl(withTheme(LifestyleButtons));

    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot for when user want to update questionnaire', () => {
    const Component = withIntl(withTheme(LifestyleButtons));

    const { container } = render(<Component questionnaireUpdateButton />);

    expect(container).toMatchSnapshot();
  });

  it('should navigate to questionnaire page', () => {
    const Component = withIntl(withTheme(LifestyleButtons));
    const { getByTestId } = render(<Component />);
    const questionnaireButton = getByTestId('btn-questionnaire');

    fireEvent.click(questionnaireButton);

    expect(navigateTo).toHaveBeenCalledWith('/questionnaire');
  });
});
