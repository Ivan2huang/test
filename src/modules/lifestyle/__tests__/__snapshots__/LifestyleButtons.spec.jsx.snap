// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifestyleButton Component should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text injectIntl(LifestyleButtons)-lifestyleBtn-1"
    data-testid="btn-questionnaire"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        style="width: 1.125rem; height: 1.125rem;"
        viewBox="0 0 18 18"
      >
        <path
          d="M17.368 1.584L16.416 0.632C15.994 0.211 15.441 0 14.889 0C14.337 0 13.784 0.211 13.362 0.632L2.954 11.041L0 18L6.959 15.046L17.368 4.638C18.211 3.795 18.211 2.427 17.368 1.584ZM6.276 14.032L2.265 15.734L3.967 11.723L12.421 3.269L14.729 5.577L6.276 14.032ZM16.519 3.789L15.578 4.73L13.27 2.422L14.211 1.481C14.392 1.3 14.633 1.2 14.889 1.2C15.145 1.2 15.386 1.3 15.567 1.481L16.519 2.433C16.893 2.807 16.893 3.415 16.519 3.789Z"
          fill="#333333"
        />
      </svg>
      <div
        class="MuiBox-root MuiBox-root-43"
      >
        Add my lifestyle data
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`LifestyleButton Component should match snapshot for when user want to update questionnaire 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text injectIntl(LifestyleButtons)-lifestyleBtn-54"
    data-testid="btn-questionnaire"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        style="width: 1.125rem; height: 1.125rem;"
        viewBox="0 0 18 18"
      >
        <path
          d="M17.368 1.584L16.416 0.632C15.994 0.211 15.441 0 14.889 0C14.337 0 13.784 0.211 13.362 0.632L2.954 11.041L0 18L6.959 15.046L17.368 4.638C18.211 3.795 18.211 2.427 17.368 1.584ZM6.276 14.032L2.265 15.734L3.967 11.723L12.421 3.269L14.729 5.577L6.276 14.032ZM16.519 3.789L15.578 4.73L13.27 2.422L14.211 1.481C14.392 1.3 14.633 1.2 14.889 1.2C15.145 1.2 15.386 1.3 15.567 1.481L16.519 2.433C16.893 2.807 16.893 3.415 16.519 3.789Z"
          fill="#333333"
        />
      </svg>
      <div
        class="MuiBox-root MuiBox-root-96"
      >
        Add my lifestyle data
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
